src/__tests__/dependency-resolution.js

```diff
@@ -383,18 +383,30 @@ tape('dependency registration with null value', t => {
 tape('dependency registration with optional deps', t => {
   const app = new App('el', el => el);
 
+  const checkString = (s: string): void => {
```

Comment:
This isn't the best developer experience for testing for Flow errors.  @KevinGrandon made a good point that we cannot currently test the failure case either.  For now, this should let us be more sound but not complete.